### PR TITLE
[BUGFIX] Chart Editor: Implement Animated Pixel Icons

### PR DESCRIPTION
**Pull request by TechnikTil**
## Does this PR close any issues? If so, link them below.
#3299 #3222 #3148 (might be more, idk)

## Briefly describe the issue(s) fixed.
This PR fixes Animated Pixel Icons showing their spritesheets by extracting the first frame of their idle from them. This is all done in-game to simplify implementation, and save storage space.

### Before:
before-image

### After:
after-image
(dont mind my fps)

## "TODO":
- [x] Extract specific frame from spritesheets.
- [x] Figure out how to load FlxFrame/FlxGraphic in a button
- [x] Figure out why the resolution is going wonk.
- [x] Change Width/Height to actual frame resolution.

### CODE REVIEW

**Review by JVNpixels — APPROVED** [review]

**Comment by TechnikTil** [comment]
I just can’t let go of this pr for some reason lol

**Comment by TechnikTil** [comment]
My life has been completed. I committed to the Funkin repository, and it wasn't a typo. I officially have 100%'d life.

**Comment by TechnikTil** [comment]
Also I was gonna implement animations but ok lol

**Comment by EliteMasterEric** [comment]
It makes more sense to just keep them on the first frame, as adding animations complicates the code for the interface and could cause things to break

**Comment by TechnikTil** [comment]
yk what good point

**Comment by TechnikTil** [comment]
ok im finished now.

File: source/funkin/play/character/CharacterData.hx [hunk 1]
```diff
@@ -281,41 +282,93 @@ class CharacterDataParser
...
+      {
+        trace('[WARN] Character ${char} has no idle in their freeplay icon.');
+        return null;
+      }
+
+      // so, haxe.ui.backend.AssetsImpl uses the parent width and height, which makes the image go crazy when rendered
+      // so this is a work around so that it uses the actual width and height
+      var imageGraphic = flixel.graphics.FlxGraphic.fromFrame(idleFrame);
+
+      var imageFrame = flixel.graphics.frames.FlxImageFrame.fromImage(imageGraphic);
```
Comment (TechnikTil) [inline source/funkin/play/character/CharacterData.hx:362]: This causes extra lag for the engine, so I made a PR to fix the issue altogether.
haxeui/haxeui-flixel#53

Comment (ianharrigan) [reply source/funkin/play/character/CharacterData.hx:362]: (this has been merged in haxeui-flixel now fyi 👍 )

Comment (TechnikTil) [reply source/funkin/play/character/CharacterData.hx:362]: yippee :D

**Comment by EliteMasterEric** [comment]
Not sure why Github doesn't like this PR but it did get merged.

**Pull request closed without merging**